### PR TITLE
correct the reference capibility for gen11 and gen12

### LINKED ISSUE #880: Ref list value is not correct on ICELake
On ICElake VDEnc, VAConfigAttribEncMaxRefFrames query return ref0 is 3 and ref1 is 0. So, B frame is not supported. But the low-delay-b is required on this entrypoint, which just converts P to B. So we should support B with a valid ref list1 and then can support low-delay-b.

### PR DESCRIPTION
**Pull request by XinfengZhang**
Fixes #880

Signed-off-by: XinfengZhang <carl.zhang@intel.com>

### CODE REVIEW

**Comment by HeJunyan** [comment]
OK, the ref list number is correct now. Please merge it.
Still need to pay attention to https://github.com/intel/media-driver/issues/990 and make the VAConfigAttribPredictionDirection correct, that may need another MR.

File: media_driver/linux/common/ddi/media_libva_caps.cpp [hunk 1]
```diff
@@ -720,7 +720,13 @@ VAStatus MediaLibvaCaps::CreateEncAttributes(
     attrib.type = VAConfigAttribEncMaxRefFrames;
     if (entrypoint == VAEntrypointEncSliceLP)
     {
+        //VDEnc Low delay P
```
Comment (Xiaogangli-intel) [inline media_driver/linux/common/ddi/media_libva_caps.cpp:723]: This should be P frame, right?

File: media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp [hunk 2]
```diff
@@ -397,6 +398,18 @@ VAStatus MediaLibvaCapsG11::GetPlatformSpecificAttrib(VAProfile profile,
...
         }
+        case VAConfigAttribPredictionDirection:
+        {
+            if(IsHevcProfile(profile) && (entrypoint == VAEntrypointEncSliceLP))
+            {
+                *value = VA_PREDICTION_DIRECTION_PREVIOUS | VA_PREDICTION_DIRECTION_BI_NOT_EMPTY;
+            }
+            else
+            {
+                *value = VA_PREDICTION_DIRECTION_PREVIOUS | VA_PREDICTION_DIRECTION_FUTURE;
```
Comment (Xiaogangli-intel) [inline media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp:409]: This is only for HEVC VME or also supports other Codecs?

File: media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp [hunk 3]
```diff
@@ -484,6 +485,12 @@ VAStatus MediaLibvaCapsG12::GetPlatformSpecificAttrib(VAProfile profile,
             }
             break;
         }
+        case VAConfigAttribPredictionDirection:
+        {
+            //HEVC gen12 support both lowdelay B and B frame
+            *value = VA_PREDICTION_DIRECTION_PREVIOUS | VA_PREDICTION_DIRECTION_FUTURE;
```
Comment (Xiaogangli-intel) [inline media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp:491]: This is only for HEVC or also supports other Codecs?

File: media_driver/linux/common/ddi/media_libva_caps.cpp (hunk 1 above)
Comment (XinfengZhang) [reply media_driver/linux/common/ddi/media_libva_caps.cpp:723]: yes, the number of reference frame in L1 is 0.

**Comment by FocusLuo** [comment]
@XinfengZhang @Sherry-Lin  The gst-vaapi HEVC regression issue is still there. for ~2 weeks 
Can you help to take this as P1?

**Comment by XinfengZhang** [comment]
https://github.com/intel/media-driver/pull/1021

**Pull request merged**